tests/test_neps_api/test_api.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+import runpy
+from pathlib import Path
+
+import pytest
+import yaml
+
+
+# To change the working directly into the tmp_path when testing function
+@pytest.fixture(autouse=True)
+def use_tmpdir(tmp_path, request):
+    os.chdir(tmp_path)
+    yield
+    os.chdir(request.config.invocation_dir)
```

Comment:
Interesting, didn't know about this kind of hack. I would advise it's better in general to request the built in `tmp_path: Path` fixture to functions that need a directory to work from, but I guess that's not easily possible for some tests.

Reply:
I took the inspiration from this test [here](https://github.com/automl/neps/blob/master/tests/test_examples.py#L9-L13).